src/ResetMap.jl

```diff
@@ -0,0 +1,287 @@
+import Base: isempty
+
+export ResetMap,
+       get_A,
+       get_b
+
+"""
+    ResetMap{N<:Real} <: LazySet{N}
+
+Type that represents a lazy reset map.
+A reset map is a special case of an affine map ``A x + b, x ∈ X`` where the
+linear map ``A`` is the identity matrix with zero entries in all reset
+dimensions, and the translation vector ``b`` is zero in all other dimensions.
+
+### Fields
+
+- `X`      -- convex set
+- `resets` -- resets (a mapping from an index to a new value)
+
+### Example
+
+```jldoctest resetmap
+julia> X = BallInf([2.0, 2.0, 2.0], 1.0);
+
+julia> r = Dict(1 => 4.0, 3 => 0.0);
+
+julia> rm = ResetMap(X, r);
+
+```
+
+Here `rm` modifies the set `X` such that `x1` is reset to 4 and `x3` is reset to
+0, while `x2` is not modified.
+Hence `rm` is equivalent to the set
+`Hyperrectangle([4.0, 2.0, 0.0], [0.0, 1.0, 0.0])`, i.e., an axis-aligned line
+segment embedded in 3D.
+
+The corresponding affine map ``A x + b`` would be:
+
+```math
+    \begin{pmatrix} 0 & 0 & 0 \\ 0 & 1 & 0 \\ 0 & 0 & 0 \end{pmatrix} x +
+    \begin{pmatrix} 4 & 0 & 0 \end{pmatrix}
+```
+
+The (in this case unique) support vector of `rm` in direction `ones(3)` is:
+
+```jldoctest resetmap
+julia> σ(ones(3), rm)
+3-element Array{Float64,1}:
+ 4.0
+ 3.0
+ 0.0
+```
+"""
+struct ResetMap{N<:Real} <: LazySet{N}
+    X::LazySet{N}
+    resets::Dict{Int, N}
+end
+
+"""
+    get_A(rm::ResetMap{N}) where {N<:Real}
+
+Return the ``A`` matrix of the affine map ``A x + b, x ∈ X`` represented by a
+reset map.
+
+### Input
+
+- `rm` -- reset map
+
+### Output
+
+The (sparse) square matrix for the affine map ``A x + b, x ∈ X`` represented by
+the reset map.
+
+### Algorithm
+
+We construct the identity matrix and set all entries in the reset dimensions to
+zero.
+"""
+function get_A(rm::ResetMap{N}) where {N<:Real}
+    n = dim(rm)
+    A = sparse(N(1)*I, n, n)
+    for i in keys(rm.resets)
+        A[i, i] = zero(N)
+    end
+    return A
+end
+
+"""
+    get_b(rm::ResetMap{N}) where {N<:Real}
+
+Return the ``b`` vector of the affine map ``A x + b, x ∈ X`` represented by a
+reset map.
+
+### Input
+
+- `rm` -- reset map
+
+### Output
+
+The (sparse) vector for the affine map ``A x + b, x ∈ X`` represented by the
+reset map.
+The vector contains the reset value for all reset dimensions, and is zero for
+all other dimensions.
+"""
+function get_b(rm::ResetMap{N}) where {N<:Real}
+    n = dim(rm)
+    b = sparsevec(Int[], N[], n)
```

Comment:
Minimalist version:

```julia
function get_b(rm::ResetMap{N}) where {N<:Real}
    return sparsevec(pairs(rm.resets), dim(rm))
end
```

Reply:
Not sure if that always works. The order in a `sparsevec` could matter.

Reply:
I think that it works because `pairs` returns `key_i => val_i`, even if it is not ordered:

```julia
julia> sparsevec([1,2], [2.0, 4.0], 3)
3-element SparseVector{Float64,Int64} with 2 stored entries:
  [1]  =  2.0
  [2]  =  4.0

julia> sparsevec([2, 1], [4.0, 2.0], 3)
3-element SparseVector{Float64,Int64} with 2 stored entries:
  [1]  =  2.0
  [2]  =  4.0
```
That said, i'm not asking to change the implementation; it was a comment after i found that it can be put in one-line using `pairs` :)

Reply:
Cool, I was not sure if the indices for `sparsevec` can be unsorted. In that case your version could even be more efficient, so I chose that.

Reply:
`pairs` is not defined in v0.6...I'll revert...

Reply:
Ouch, we still live in the past...